pkg/standalone/standalone.go

```diff
@@ -50,8 +51,64 @@ const (
 	DaprRedisContainerName = "dapr_redis"
 )
 
+func isInstallationRequired(installLocation, requestedVersion string) bool {
+	destDir := daprDefaultLinuxAndMacInstallPath
+	if installLocation == "" {
+		if runtime.GOOS == daprWindowsOS {
+			destDir = daprDefaultWindowsInstallPath
+		}
+	}
+
+	daprdBinaryPath := filepath.Join(destDir, daprRuntimeFilePrefix) //e.g. /usr/local/bin/daprd or c:\\daprd
+
+	//First time install?
+
+	_, err := os.Stat(daprdBinaryPath)
+	if os.IsNotExist(err) {
+		return true
+	}
+
+	var msg string
+
+	//what's the installed version?
+	v, err := utils.RunCmdAndWait(daprdBinaryPath, "--version")
+	if err != nil {
+		msg = fmt.Sprintf("unable to determine installed Dapr version at %s", destDir)
+		print.FailureStatusEvent(os.Stdout, msg)
+		return false
+	}
+	installedVersion := strings.TrimSpace(v)
+
+	//"latest" version requested. need to check the corresponding version
+	if requestedVersion == daprLatestVersion {
+		latestVersion, err := getLatestRelease(daprGitHubOrg, daprGitHubRepo)
+		if err != nil {
+			msg = fmt.Sprintf("latest Dapr version information could not be found - %s", err.Error())
+			print.FailureStatusEvent(os.Stdout, msg)
+			return false
+		}
+		latestVersion = latestVersion[1:]
+		if installedVersion == latestVersion {
+			requestedVersion = latestVersion
+		}
+	}
+
+	//if daprd exists, need to confirm if the intended version is same as the current one
+	if installedVersion == requestedVersion {
+		msg = fmt.Sprintf("required version %s is the same as installed version at - %s", requestedVersion, destDir)
+		print.InfoStatusEvent(os.Stdout, msg)
+		return false
+	}
+
+	return true
+}
+
 // Init installs Dapr on a local machine using the supplied runtimeVersion.
 func Init(runtimeVersion string, dockerNetwork string, installLocation string) error {
+	//confirm if installation is needed
+	if !isInstallationRequired(installLocation, runtimeVersion) {
+		return errors.New("installation will not proceed")
```

Comment:
Rather than error, Should it just say somthing along the lines of "latest version is already installed, it will not be overwritten"

Reply:
updated: simply returns `nil` (no error). the `isInstallationRequired` function will already provide the info message to the user so explicit logging is not needed here